src/tools/illink/Directory.Build.targets

```diff
@@ -1,33 +1,14 @@
 <Project>
-  <Import Project="Sdk.targets" Sdk="Microsoft.DotNet.Arcade.Sdk" />
+  <Import Project="..\..\..\Directory.Build.targets" />
 
   <Import Project="$(_ToolsProjectTargets)" Condition="Exists('$(_ToolsProjectTargets)')" />
 
   <Target Name="VSTestIfTestProject">
     <CallTarget Targets="VSTest" Condition="'$(IsTestProject)' == 'true'" />
   </Target>
 
-
-  <!-- Shared logic for reference assemblies. This enables apicompat and
-       packaging logic for projects that have corresponding reference assemblies. -->
-  <PropertyGroup>
-    <ContractProject Condition="'$(ContractProject)' == ''">$(MSBuildProjectDirectory)\ref\$(MSBuildProjectName).csproj</ContractProject>
-    <HasMatchingContract Condition="Exists('$(ContractProject)')">true</HasMatchingContract>
-    <RunApiCompat Condition="'$(HasMatchingContract)' != 'true'">false</RunApiCompat>
-    <TargetsForTfmSpecificContentInPackage Condition="'$(HasMatchingContract)' == 'true'">$(TargetsForTfmSpecificContentInPackage);_AddReferenceAssemblyToPackage</TargetsForTfmSpecificContentInPackage>
-  </PropertyGroup>
-  <!-- Work around ApiCompat breaking change https://github.com/dotnet/arcade/issues/5361 -->
-  <ItemGroup Condition="'$(RunApiCompat)' == 'true'">
-    <_DependencyDirectories Include="$(IntermediateOutputPath)" />
-  </ItemGroup>
-  <!-- Add a ProjectReference to the reference assembly project for ApiCompat. -->
-  <!-- This also ensures that it is built when the current project gets packaged. -->
-  <ItemGroup Condition="'$(HasMatchingContract)' == 'true'">
-    <ProjectReference Include="$(ContractProject)">
-      <ReferenceOutputAssembly>false</ReferenceOutputAssembly>
-      <OutputItemType>ResolvedMatchingContract</OutputItemType>
-    </ProjectReference>
-  </ItemGroup>
+  <!-- Import reference assembly and ApiCompat logic --> 
+  <Import Project="$(RepositoryEngineeringDir)resolveContract.targets" /> 
```

Comment:
Same as in libraries, this should be only be imported for source projects and not for reference assemblies or test projects: https://github.com/dotnet/runtime/blob/d45904acf566817af32eb20a3cbeff042c0e717d/src/libraries/Directory.Build.targets#L128